src/__template__/ComponentTemplate.vue

```diff
@@ -90,6 +101,70 @@ const rootElementStyles = computed((): Record<string, string> => ({
   lineHeight: '0',
   width: iconSize.value,
 }))
+
+/**
+ * Prefix all ids within the SVG element and update all corresponding references to these ids. This is useful to avoid id conflicts when utilizing multiple instances of the same SVG in the DOM.
+ *
+ * @param {SVGElement} svgElement - The SVG element whose potential IDs need to be prefixed.
+ */
+const prefixSvgIds = (svgElement: SVGElement): void => {
+  if (!svgElement) {
+    return
+  }
+
+  const defsElement = svgElement.querySelector('defs')
+
+  // Prepare a map to store the original and new IDs for quick lookups
+  const idMap: Record<string, string> = {}
+
+  if (defsElement) {
+    // Prefix all direct children of <defs> that have an `id`
+    defsElement.querySelectorAll('[id]').forEach((element) => {
+      const originalId = element.getAttribute('id')
+      const newId = `${Math.random().toString(36).substring(2, 12)}-${originalId}`
+
+      // Map old id to new id
+      idMap[originalId!] = newId
+
+      // Update the element's id with the new prefixed id
+      element.setAttribute('id', newId)
+    })
+  }
```

Comment:
I think we can also reference elements outside a `<defs>`?

Reply:
I don't think there's a reason to? Would there ever be a referenced id outside of the defs element?

Reply:
Refactored 👍🏼